js/app/application.js

```diff
@@ -227,6 +253,25 @@ const Application = new Knowledge.Class({
         });
     },
 
+    QueryNotifications: function() {
+        return [];
+    },
+
+    ContentQueryAsync: function (query, invocation) {
+        let engine = Eknc.Engine.get_default();
+        engine.query_promise(Eknc.QueryObject.new_from_props({
+            tags_match_all: ['EknArticleObject'],
+            limit: 5
+        })).then(results =>
+            invocation.return_value(new GLib.Variant('(s)', [JSON.stringify(results.models.map(m => ({
```

Comment:
I really don't like marshalling all of this through JSON. I would return variants directly, like the search provider interface does.